### ISSUE TITLE
Android / iOS - After regaining network user is redirected from manage members page to LHN

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#7549]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Launch the app
2. Log in with expensiafail account
3. Go to Setting - Members - Manage Members
4. Kill the internet connection
5. Click on invite and enter a user address (you can access to
6. Select the member and click on invite
7. Verify the member is added to the list
8. Go Back online

## Expected Result:
Verify the new user is still displayed on the list of Manage Members

## Actual Result:
Manage member page leads to LHN page after back to Online

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS
- Android

**Version Number:** 1.1.35 -0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/152379334-5d2d811f-6a11-40ac-ab88-d5cbd76aec51.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @neil-marcellini (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @dylanexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by neil-marcellini** [comment]
I was not able to reproduce this on my IOS simulator. But on my physical IOS device on production I found something very interesting. If you follow the steps and then press invite only once before hitting the back button, when you come back online you will be navigated back 1 screen, to the workspace screen. If you press invite twice you will be navigated back 2 screens when you come back online, to the settings screen. If you press invite 3 or more times then you will be navigated all the way back to the LHN / chats page as is shown in the video. I'll make this a monthly issue since it only happens if you go offline and come back online in a particular way. It is an obscure bug but still something that we should fix.

**Comment by parasharrajat** [comment]
The `navigation.goback()` callback is called on each API response which is received when the internet is back. This will be fixed soon in one of the other issues.

**Comment by dylanexpensify** [comment]
not overdue

**Comment by dylanexpensify** [comment]
Any update here @parasharrajat ?

**Comment by parasharrajat** [comment]
Sorry @dylanexpensify, I am not tracking it as it is not assigned to me.

**Comment by dylanexpensify** [comment]
ah sorry @parasharrajat my misread! @neil-marcellini any update here? Do we need this posted to upwork yeah?

**Comment by neil-marcellini** [comment]
> The navigation.goback() callback is called on each API response which is received when the internet is back. This will be fixed soon in one of the other issues.

I don't know what issue @parasharrajat was referring to here. Are you referring to [this call here](https://github.com/Expensify/App/blob/173d234f0fb4eaca03c315a03237836d9d5ebad8/src/pages/workspace/WorkspaceBankAccountPage.js#L75)? I am still able to reproduce this on mobile web on production. I haven't tested on IOS but I bet it occurs there too. Yes, please create an upwork job for this.

**Comment by parasharrajat** [comment]
Oh, I didn't link the PR :scream:. That PR is discontinued due to unexpected circumstances and the issue was marked on HOLD for offline design. https://github.com/Expensify/App/issues/6676. 


So yeah, please create a job for this.

@neil-marcellini I was referring to this https://github.com/Expensify/App/blob/173d234f0fb4eaca03c315a03237836d9d5ebad8/src/libs/actions/Policy.js#L338. 

After use regain connection and he has not refreshed the page, API calls will complete and the success callback will fire with 200. As a result, this will be called and causing the user to go back from any back. 
This is a very good example of adding side effects to the functions. All of this should be handled by the Component itself, not by the action. Navigation is a kind of action. And this calling an action from another action is fine when only modifies the state. But this is creating a side-effect and it will cause many issues.

**Comment by dylanexpensify** [comment]
WOW, missed this completely, my bad! 

Internal: https://www.upwork.com/ab/applicants/1521837452385910784/job-details
External: https://www.upwork.com/jobs/~01cec91a17c169473e

**Comment by dylanexpensify** [comment]
Also, reassigning as headed on parental leave next week!!

**Comment by marcochavezf** [comment]
There're ongoing changes related to the network logic and probably this issue will be fixed when the Offline UX Patterns are implemented. I think we should close this issue atm and maybe re-open it if it's still happening after implementing the Offline Patterns, @stephanieelliott could you remove the Upwork jobs?
